xrootd/client/filesystem.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2018 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package client // import "go-hep.org/x/hep/xrootd/client"
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/pkg/errors"
+	"go-hep.org/x/hep/xrootd/protocol"
+	"go-hep.org/x/hep/xrootd/protocol/dirlist"
+)
+
+// Filesystem contains filesystem-related methods of the XRootD protocol.
```

Comment:
to piggy-back on the stdlib (and on `go-fuse`):
```
$> go doc net/http.FileSystem 
type FileSystem interface {
	Open(name string) (File, error)
}
    A FileSystem implements access to a collection of named files. The elements
    in a file path are separated by slash ('/', U+002F) characters, regardless
    of host operating system convention.

```

I would `s/Filesystem/FileSystem/`